src/meta/src/manager/scale.rs

```diff
@@ -0,0 +1,245 @@
+// Copyright 2022 Singularity Data
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::VecDeque;
+use std::sync::atomic::{AtomicU32, Ordering};
+use std::sync::Arc;
+
+use log::error;
+use risingwave_common::error::{ErrorCode, Result};
+use risingwave_pb::meta::scale_task::TaskType;
+use risingwave_pb::meta::{ScaleTask, TaskStatus};
+use tokio::sync::{Mutex, Semaphore};
+
+use crate::model::{MetadataModel, ScaleTaskId};
+use crate::storage::MetaStore;
+
+pub type ScaleManagerRef<S> = Arc<ScaleManager<S>>;
+
+/// [`ScaleManager`] is used to receive scale tasks.
+#[allow(dead_code)]
+pub struct ScaleManager<S: MetaStore> {
+    meta_store: Arc<S>,
+
+    next_task_id: AtomicU32,
```

Comment:
How about make it a simple uuid? This field will be reset when meta restart.

Reply:
I don't understand what the simple uuid means. We can only use & reference, so we have to choose `AtomicU32` or `Mutex<u32>`.

Reply:
I mean that you can change the task id to `String` and use `uuid::Uuid` to generate it.

Reply:
+1 for UUID, which matches the behavior of all other tasks in our system. There's no need to store anything here, we can generate a new UUID in place we schedule a task.

Reply:
Since a task queue introduced here, we might have several pending tasks exist here. When recovery from meta store, all pending tasks should be add into queue in order. I still suggest to use UUID here and add a `create_time` for every task, which can be used for recovery and also meaningful for like tracing or dashboard etc in the future.